executor_bot/src/dex/openbook.ts

```diff
@@ -0,0 +1,141 @@
+import { Market, OpenOrders } from "@project-serum/serum";
+import { TOKEN_PROGRAM_ID } from "@raydium-io/raydium-sdk";
+import {
+  PublicKey,
+  AccountMeta,
+  SYSVAR_RENT_PUBKEY,
+  Connection,
+  Transaction,
+  Keypair,
+  sendAndConfirmTransaction,
+} from "@solana/web3.js";
+import BN from "bn.js";
+
+export const OPENBOOK_V3_PROGRAM_ID = new PublicKey(
+  "srmqPvymJeFKQ4zGQed1GFppgkRHL9kaELCbyksJtPX"
+);
+
+export const openBookTradeAccounts = async (
+  serumMarket: Market,
+  collateralAccount: PublicKey,
+  tradeDestinationAccount: PublicKey,
+  openOrdersKey: PublicKey,
+  openOrdersOwner: PublicKey
+): Promise<AccountMeta[]> => {
+  const vaultSigner = await deriveVaultSigner(serumMarket);
+  return [
+    { pubkey: serumMarket.programId, isWritable: false, isSigner: false },
+    { pubkey: serumMarket.address, isWritable: true, isSigner: false },
+    { pubkey: serumMarket.bidsAddress, isWritable: true, isSigner: false },
+    { pubkey: serumMarket.asksAddress, isWritable: true, isSigner: false },
+    { pubkey: openOrdersKey, isWritable: true, isSigner: false },
+    {
+      // @ts-ignore
+      pubkey: serumMarket._decoded.requestQueue,
+      isWritable: true,
+      isSigner: false,
+    },
+    {
+      // @ts-ignore
+      pubkey: serumMarket._decoded.eventQueue,
+      isWritable: true,
+      isSigner: false,
+    },
+    {
+      // @ts-ignore
+      pubkey: serumMarket._decoded.baseVault,
+      isWritable: true,
+      isSigner: false,
+    },
+    {
+      // @ts-ignore
+      pubkey: serumMarket._decoded.quoteVault,
+      isWritable: true,
+      isSigner: false,
+    },
+    { pubkey: vaultSigner, isWritable: false, isSigner: false },
+    { pubkey: TOKEN_PROGRAM_ID, isWritable: false, isSigner: false },
+    { pubkey: SYSVAR_RENT_PUBKEY, isWritable: false, isSigner: false },
+    // This is the SRM referral account
+    { pubkey: SYSVAR_RENT_PUBKEY, isWritable: false, isSigner: false },
+    { pubkey: openOrdersOwner, isWritable: false, isSigner: false },
+    { pubkey: collateralAccount, isWritable: false, isSigner: false },
+    { pubkey: tradeDestinationAccount, isWritable: true, isSigner: false },
+  ];
+};
+
+export const deriveVaultSigner = async (
```

Comment:
Duplicate of `@mithraic-labs/poseidon` `OpenBookDex.deriveVaultSigner`. Lets DRY it up